[Technical review] BiDi - New pages for browsingContext module commands: captureScreenshot, handleUserPrompt, print, setViewport (#44931)
mdn/content: files/en-us/web/webdriver/reference/bidi/modules/browsingcontext/capturescreenshot/index.md

page-type: webdriver-command
browser-compat: webdriver.bidi.browsingContext.captureScreenshot
sidebar: webdriver
---

The `browsingContext.captureScreenshot` [command](/en-US/docs/Web/WebDriver/Reference/BiDi/Modules#commands) of the [`browsingContext`](/en-US/docs/Web/WebDriver/Reference/BiDi/Modules/browsingContext) module captures an image of the specified [context](/en-US/docs/Web/WebDriver/Reference/BiDi/Modules/browsingContext#contexts) and returns it as a [Base64](/en-US/docs/Glossary/Base64)-encoded string.

## Syntax

```json-nolint
/* With required parameters */
{
  "method": "browsingContext.captureScreenshot",
  "params": {
    "context": "93ee5bd6-d256-4608-a002-9a8995cc0e5f"
  }
}

/* With required and optional parameters */
{
  "method": "browsingContext.captureScreenshot",
  "params": {
    "context": "93ee5bd6-d256-4608-a002-9a8995cc0e5f",
    "origin": "document",
    "format": {
      "type": "image/jpeg",
      "quality": 0.8
    },
    "clip": {
      "type": "box",
      "x": 10,
      "y": 10,
      "width": 200,
      "height": 100
    }
  }
}
```

### Parameters

The `params` field contains:

- `clip` {{optional_inline}}

>   - : An object that specifies whether to limit the image capture to an area or an element.

Reviewer: ```suggestion
  - : An object that limits the capture to a rectangular area or to a single element.
```

    If `clip` is not specified, the entire `origin` area is captured.
    It contains the following fields:

    - `type`
      - : A string that specifies whether the screenshot area is a rectangular region or a specific element.

>         It can take one of the following two values:

Reviewer: ```suggestion
        It can take one of the following values:
```

        - `"box"`
          - : Captures a rectangular region.
            The area is specified using coordinates that are offsets from the top-left corner of `origin`.
            The `clip` object has the following fields for `type: "box"`:

            - `height`
              - : A number that specifies the height of the rectangle.
            - `width`
              - : A number that specifies the width of the rectangle.
            - `x`
              - : A number that specifies the horizontal offset of the rectangle from the top-left corner of `origin`.
            - `y`
              - : A number that specifies the vertical offset of the rectangle from the top-left corner of `origin`.
        - `"element"`
          - : Captures the bounding box of a specific element.

>             The clip object also has the following field for `type: "element"`:

Reviewer: ```suggestion
            The `clip` object also has the following field for `type: "element"`:
```

- `context`
  - : A string that contains the ID of the context to capture as a screenshot.
    Context IDs are returned by commands such as [`browsingContext.getTree`](/en-US/docs/Web/WebDriver/Reference/BiDi/Modules/browsingContext/getTree).
- `format` {{optional_inline}}
  - : An object that specifies the image encoding to use. If not specified, the screenshot is encoded as `"image/png"`.
    It contains the following fields:
    - `quality` {{optional_inline}}
      - : A number between `0.0` and `1.0` that specifies the compression quality to use for [lossy formats](/en-US/docs/Glossary/Lossy_compression) such as `"image/jpeg"`.
        If not specified, the compression level used is browser-dependent.
    - `type`
      - : A string that contains the image [MIME type](/en-US/docs/Glossary/MIME_type) to encode the screenshot as, for example `"image/png"` or `"image/jpeg"`.
- `origin` {{optional_inline}}
  - : A string that specifies the area for the screenshot.
    It can take one of the following values:
    - `"document"`: The area spans the entire scrollable document, including content outside the visible viewport.

>     - `"viewport"`: The area is the [visible viewport](/en-US/docs/Glossary/Visual_Viewport). This is the default.

Reviewer: ```suggestion
    - `"document"`: The area spans the entire scrollable document, including content outside the [visible viewport](/en-US/docs/Glossary/Visual_Viewport).
    - `"viewport"`: The area is the visible viewport. This is the default.
```

I would link the first occurrence.

Author: Agree, fixed

### Return value

The `result` object in the response contains the following fields:

- `data`
  - : A string that contains the Base64-encoded image data.

### Errors

- [`invalid argument`](/en-US/docs/Web/WebDriver/Reference/Errors/InvalidArgument)
  - : A required parameter is missing or has an invalid type.

- `no such frame`
  - : No context with the given context ID is found.

- `no such element`

>   - : The DOM element referenced by `clip.element` cannot be resolved or does not belong to the document of the context being captured.

Reviewer: ```suggestion
- `no such element`
  - : The DOM element referenced by `clip.element` cannot be resolved or does not belong to the document of the context being captured.

- `no such frame`
  - : No context with the given context ID is found.
```

To keep the alphabetic order
